### PR TITLE
fix: make buttons look like further buttons

### PR DESCRIPTION
**Pull request by zz-james**
ticket: https://pi-top.atlassian.net/browse/OS-1146

replace styled buttons with primary button component from Further, leave unstyled (link) buttons the same.

changed the margin on the dialog to match design on ticket. 

My editor made some formatting changes, sorry about that - are we all on the same settings? it would be a pain if we kept automatically reformatting each other... 

oh and then there was the tests. It feels bad that such a simple thing as testing the disabled state of a button becomes a tiny bit harder with these fancy buttons but I don't know any way around it... button element is now the parent of the element with the text of the button in, so had to add .parentElement to all the getByText()s in the tests

### CODE REVIEW

**Review by olivierwilkinson — CHANGES_REQUESTED** [review]
I've got an off by one issue on Firefox, Safari and Chrome, they are all a little different but the problem seems to be the same:

Screenshot 2022-02-04 at 15 50 02

**Comment by zz-james** [comment]
I've improved it (at least in my experience tested mac, win10 and ubuntu) - the outline is very slightly thicker than it was in the original design because the original had a stroke width of 1.8, which caused a bit of, I guess dithering? to approximate something wider than 1 but lass then 2, now it has a solid 2 which improves the situation a lot, especially around the anti-aliasing at the ends of the curves. 

even after extensive refactoring FireFox on mac (but not on Windows or Ubuntu) still was off by  one, so now it has it's own CSS to fix it.

**Comment by olivierwilkinson** [comment]
I'm finding issues when the font loads in, but before that it looks perfect.
Before font loads in
Screenshot 2022-02-09 at 11 33 30
After font loads in
Screenshot 2022-02-09 at 11 31 20

**Comment by zz-james** [comment]
unhelpfully I wasn't seeing this even using the Roboto font. I've made some changes that might help.

**Comment by angusjfw** [comment]
this looks perfect on my pi-top screen and in windows if I use 100 or 200% ui scaling. With my regular 175% it's fine too just a little stubbly:
image
image

also there is a snapshot failure

**Comment by zz-james** [comment]
I've taken a couple of screenshots of Chrome at 175% (by far the stubbliest) you are right it is better in the old code, here's what's happening, this is a screenshot at 175% then zoomed 400% in Gimp

buttons_zoomed


I'm finding it difficult to recreate that part where the straight top/bottom is actually 3 px wide with one px being a semi opaque, and if there curves are going to meet the straights without any hair than either the straights need to have this or the curves need to loose their semi-opaque parts... I haven't figured out this level of control.

For now, I am not upset at all to leave the Further buttons how they are, I guess there's pros and cons

**Comment by zz-james** [comment]
I found a way to control the stubble pixels using a clip-path, surrounding the button with a rect that acts as a mask - this trims the bottom and the top so there's no wayward anti-aliasing - this screenshot is 175% on browser then zoomed to 400
Screenshot from 2022-02-10 09-06-01

I made a few other little changes to make the roll-over fill fit into the shape better 

commit to follow shortly

File: node_modules/.yarn-integrity [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
[the remark this answers is not in the document]
Comment (zz-james) [reply node_modules/.yarn-integrity:1]: uh no idea when that arrived I didn't ask for it. I will remove

File: frontend/src/components/atoms/dialog/Dialog.module.css [hunk 2]
```diff
@@ -47,7 +47,7 @@
   padding: 0 40px 60px 40px;
   font-weight: 200;
   font-size: 16px;
-  margin: 0;
+  margin: 0 50px;
```
[the remark this answers is not in the document]
Comment (zz-james) [reply frontend/src/components/atoms/dialog/Dialog.module.css:50]: oh weird I thought it was needed to make the dialog text like the design, but will remove :-)

**Comment by angusjfw** [comment]
There's been a bit of back and forth with this isSingleOption ValueType issue that's currently breaking the build.
https://github.com/pi-top/pi-topOS-Web-Portal/commit/13b7e8fc3102f368c3f63caa4ec2eccac79313cc

It seems like maybe this was a breaking change in react-select, between 3.0 and 3.1, without being marked as a major update. You're yarn lock is now resolving to 3.1 again which is why this broke again I think.

Maybe just try bumping react-select to the latest, 5.2.2? Or if that's problematic I suppose just revert this change again.

**Comment by zz-james** [comment]
ok have replaced the original yarn.lock file from master as there were more problems with react-testing-library building ok now.

**Pull request merged**